templates/components/form/form.jsx

```diff
@@ -25,30 +27,47 @@ var Form = React.createClass({
     return {
       username: null,
       password: null,
-      email: null
+      email: null,
+      checked: false,
+      dirty: {}
     };
   },
   buildFormElement: function(key, i) {
-    var id = Object.keys(this.props.fields[i]);
+    // we always expect this.props.fields[i] to be one object with one property.
+    var self = this;
+    var id = Object.keys(this.props.fields[i])[0];
     var placeholder = this.props.fields[i][id].placeholder;
     var type = this.props.fields[i][id].type;
     var label = this.props.fields[i][id].label;
     var labelPosition = this.props.fields[i][id].labelPosition;
-    var className = this.getIconClass(id);
+    var errorMessage = this.props.fields[i][id].errorMessage;
+    var isDisabled = this.props.fields[i][id].disabled;
+    var focus = this.props.fields[i][id].focus;
+    var username = this.props.fields[i][id].username;
+    this.passChecked = this.props.fields[i][id].checked;
 
-    function foo() {
-      console.log('foo');
-      this.handleValidation(id);
+    function dirty(err, valid) {
+      var state = self.state.dirty;
```

Comment:
you're not extracting "state", but the thing that records which elements are dirty. `var dirty` ?